appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -1031,11 +1031,18 @@ private boolean insertObjectReference(ApiContext context, Reference<?> referee,
             return false;
         }
 
+        final Schema schema = AnnotationInfo.valueOf(referenceClass).getAnnotation(Schema.class);
+
         // Set the reference name
-        referee.setRef(referenceClass.getSimpleName());
+        if (schema == null || schema.name().isEmpty()) {
+            referee.setRef(referenceClass.getSimpleName());
+        } else {
+            referee.setRef(schema.name());
+        }
 
         // Create the schema
         visitSchema(AnnotationInfo.valueOf(referenceClass).getAnnotation(Schema.class), referenceClass, context);
+        visitSchema(schema, referenceClass, context);
```

Comment:
The `visitSchema()` is now called twice with the same parameters.

Reply:
Thank you. That line is completely in vain there. Removed in https://github.com/payara/Payara/pull/4494/commits/369794a7181575d204db98aaad315e623ae661d8